[Build] Upgrade Gradle to 8.2.1 and AGP to 8.1.0

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

This PR upgraded Gradle to [8.2.1](https://docs.gradle.org/8.2.1/release-notes.html) and AGP to [8.1.0](https://developer.android.com/build/releases/gradle-plugin#8-1-0).

FYI: The [8.1.1](https://docs.gradle.org/8.1.1/release-notes.html) Gradle upgrades had to happen in 2 parts (87b30ea64066817ffb76fc3378f7fb8d12c81581 and d6e0d5b944be4eb1be9613e81c0b18fe4b18206b). This is because of the breaking changes that this upgrade introduces, including the `AGP` upgrade and its associated breaking changes (namespace, build option, etc).

-----

🚫 Blocked By:

- [x] `Android Gradle plugin requires Java 17 to run. You are currently using Java 11.`

<details>
    <summary>CI Failure</summary>

```
[2023-07-17T11:47:46Z] FAILURE: Build failed with an exception.
--
  | [2023-07-17T11:47:46Z]
  | [2023-07-17T11:47:46Z] * Where:
  | [2023-07-17T11:47:46Z] Build file '/var/lib/buildkite-agent/builds/ci-android-i-03301d03eb1c8e4e5-1/automattic/woocommerce-android/quicklogin/build.gradle' line: 2
  | [2023-07-17T11:47:46Z]
  | [2023-07-17T11:47:46Z] * What went wrong:
  | [2023-07-17T11:47:46Z] An exception occurred applying plugin request [id: 'com.android.test']
  | [2023-07-17T11:47:46Z] > Failed to apply plugin 'com.android.internal.test'.
  | [2023-07-17T11:47:46Z]    > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
  | [2023-07-17T11:47:46Z]       Your current JDK is located in /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.amzn2.0.1.x86_64
  | [2023-07-17T11:47:46Z]       You can try some of the following options:
  | [2023-07-17T11:47:46Z]        - changing the IDE settings.
  | [2023-07-17T11:47:46Z]        - changing the JAVA_HOME environment variable.
  | [2023-07-17T11:47:46Z]        - changing `org.gradle.java.home` in `gradle.properties`.
```

</details>

-----

Breaking Change List:

1. [Enable build config build feature for quicklogin and woocommerce](https://github.com/woocommerce/woocommerce-android/commit/db2340bb8a5fa688f29cc35b61f8d5d067192cbe)
2. [Update google services plugin to 4.3.15](https://github.com/woocommerce/woocommerce-android/commit/5966c09c3907bd625e3bd12c582711ce6aebf319)
3. [Resolve all violating non final res references for main](https://github.com/woocommerce/woocommerce-android/commit/2e77a0c9ee5f6e97699bbc53f04c519591b7e5eb)
4. [Resolve all violating non final res references for android test](https://github.com/woocommerce/woocommerce-android/commit/c7147dd2618cf10f5a0fa5dae429818007106925)
5. [Replace dependency substitution substitute with method with using](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/d9ed4bb9b982ad979cd4ce63731e2a0f43fc02e3)

Test Resolution List:

1. [Fix gson related test failures on site picker view model test](https://github.com/woocommerce/woocommerce-android/commit/9b3cf55e00d60b3fc14fcab07e9967abf5a058dd)

Lint Suppression List:

1. [Suppress new api lint warning for context ext](https://github.com/woocommerce/woocommerce-android/commit/86e64288484ef27892771a86eb1cb5f677207bd1)

Dependency Versions Refactor List:

1. [Extract google services plugin related version](https://github.com/woocommerce/woocommerce-android/commit/f2c07a94c644c1c1720517da25b4a4d2a1dd2be3)
2. [Extract wzieba tracks plugin related version](https://github.com/woocommerce/woocommerce-android/commit/204a619363973ce4402472c3faaede68f4482f2f)
6. [Extract sentry plugin related version](https://github.com/woocommerce/woocommerce-android/commit/ef4e67592972b16e8b871dec6829d674a911b818)
7. [Extract violation comments plugin related version](https://github.com/woocommerce/woocommerce-android/commit/b71e2222424f2628a2d7f68afc0b46fd59b230a0)

-----

## Associated libraries

You can reference the `[Build] Upgrade Gradle to 8.1.1 and AGP to 8.0.2` PRs of the below associated libraries for testing/verification purposes:

- [x] [MediaPicker#69](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/69)
- [x] [Login#119](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/119)
- [x] [FluxC#2772](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2772)

-----

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

1. Verify that all the CI checks are successful.
2. Smoke test the app, without composite builds.
3. Smoke test the app again, but this time, using the composite builds for all 3 libraries (following the instructions in [local-builds.gradle-example](https://github.com/woocommerce/woocommerce-android/blob/trunk/local-builds.gradle-example) file). PS: Make sure you're on the `Gradle & AGP` branches on all 3 libraries.

-----

### Testing instructions

- [x] Update all 3 library references to PR hash (this is done for testing purposes):
  - [x] [mediapickerVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/ee964891667eaa30bf774c0225f20980779e9832) (v1) + [mediapickerVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/75bc96949c64214a7c62f2236a243903e7eabba9) (v2)
  - [x] [wordPressLoginVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/d692875c38b7978f94cc1058bd09d0a1cc767683) (v1) + [wordPressLoginVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/d7317feb17a5641aa435dd60d0c2cf5b692b8cbe) (v2)
  - [x] [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/7aee36cd457296215fec3eda240b5dd29af54275) (v1) + [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/732f43d9c1892b29add62bf11fd44cfae28dda44) (v2) + [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/9ea0b97db6d7a452df250bb753a4fc5c4a1cc6ab) (v3)
- [x] Mark PR as `Ready for review`.
- [x] Merge all 3 library PRs into their `trunk` (after them being code reviewed and approved)
  - [x] [MediaPicker#69](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/69)
  - [x] [Login#119](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/119)
  - [x] [FluxC#2772](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2772)
- [x] Update all 3 library references to `trunk` hash or tag:
  - [x] [mediapickerVersion](TODO)
  - [x] [wordPressLoginVersion](TODO)
  - [x] [fluxCVersion](TODO)
- [x] Wait for PR to get approved.
    - [x]  TODO: Test all the PRs together once done with the reviews (@oguzkocer).
- [x] Remove `status: do not merge` label.
- [x] Merge this PR.

-----

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
Build: Upgrade gradle to 8.1.1 (1/2)

Release Notes: https://docs.gradle.org/8.1.1/release-notes.html
Command: ./gradlew wrapper --gradle-version=8.1.1
--distribution-type=all

------------------------------------------------------------------------

FYI: You will notice that the associated 'gradle-wrapper.jar', 'gradlew'
and 'gradlew.bat' files are not updated. This is because of the breaking
changes that this upgrade introduces, including the 'AGP' upgrade and
its associated breaking changes (namespace, build option, etc). As such,
this Gradle upgrade command is only half done.

PS: When all the breaking changes are resolved, this Gradle command will
be issued again, which should fully complete this upgrade.

ParaskP7 committed:
Build: Upgrade agp to 8.0.2

Release Notes: https://developer.android.com/build/releases/
gradle-plugin#agp-8-0-2

ParaskP7 committed:
Build: Enable build config build feature for quicklogin and woocommerce

This is an AGP version '8.0' breaking change that changes the build
option default values, see build failure below:

------------------------------------------------------------------------

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':quicklogin'.
> com.android.builder.errors.EvalIssueException: defaultConfig contains
 custom BuildConfig fields, but the feature is disabled.

------------------------------------------------------------------------

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':WooCommerce'.
> com.android.builder.errors.EvalIssueException: Build Type 'debug'
 contains custom BuildConfig fields, but the feature is disabled.

------------------------------------------------------------------------

Explanation: "Starting with AGP 8.0, the default values for these flags
have changed to improve build performance. AGP 8.0 doesn't generate
'BuildConfig' by default. You need to specify this option using the DSL
in the projects where you need it."

For more info see: https://developer.android.com/build/releases/
gradle-plugin#default-changes

ParaskP7 committed:
Build: Update google services plugin to 4.3.15

Release Notes: https://firebase.google.com/support/release-notes/android
#google-services_plugin_v4-3-15

This update is necessary because the build is otherwise failing with the
below 'GoogleServicesTask' related problem, which seems to be an
incompatibility of some shorts, that is related to the newer version of
AGP (8.0):

------------------------------------------------------------------------

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task
 ':WooCommerce:processWasabiDebugGoogleServices'
  (type 'GoogleServicesTask').
  - Gradle detected a problem with the following location:
   '/Users/.../WooCommerce/build/generated/res/google-services/wasabi/
    debug'.

    Reason: Task ':WooCommerce:mergeWasabiDebugResources' uses this
    output of task
     ':WooCommerce:processWasabiDebugGoogleServices' without declaring
      an explicit or implicit dependency. This can lead to incorrect
      results being produced, depending on what order the tasks are
      executed.

    Possible solutions:
      1. Declare task ':WooCommerce:processWasabiDebugGoogleServices' as
       an input of ':WooCommerce:mergeWasabiDebugResources'.
      2. Declare an explicit dependency on
       ':WooCommerce:processWasabiDebugGoogleServices' from
       ':WooCommerce:mergeWasabiDebugResources' using Task#dependsOn.
      3. Declare an explicit dependency on
       ':WooCommerce:processWasabiDebugGoogleServices' from
       ':WooCommerce:mergeWasabiDebugResources' using Task#mustRunAfter.

    Please refer to https://docs.gradle.org/8.1.1/userguide/
     validation_problems.html#implicit_dependency for more details about
     this problem.

------------------------------------------------------------------------

FYI.1: The current '4.3.10' version is almost 2 years old (09/08/21):
- https://firebase.google.com/support/release-notes/android
#google-services_plugin_v4-3-10

FYI.2: The latest '4.3.15' version went live half a year ago (19/01/23),
which improved support for 'camelCase' variant and build type names.:
- https://firebase.google.com/support/release-notes/android
#google-services_plugin_v4-3-15

FYI:3: See also:
- '4.3.14', which improved task caching and awareness of changes in
'google-services.json' files.
https://firebase.google.com/support/release-notes/android
#google-services_plugin_v4-3-14
- '4.3.13', which fixed unintended binary incompatibility with the
Crashlytics Gradle plugin.
https://firebase.google.com/support/release-notes/android
#google-services_plugin_v4-3-13
- '4.3.12', which Improved compatibility with AGP 7.1+.
https://firebase.google.com/support/release-notes/android
#google-services_plugin_v4-3-12

ParaskP7 committed:
Deps: Extract google services plugin related version

ParaskP7 committed:
Deps: Extract wzieba tracks plugin related version

ParaskP7 committed:
Deps: Extract sentry plugin related version

ParaskP7 committed:
Deps: Extract violation comments plugin related version

ParaskP7 committed:
Build: Resolve all violating non final res references for main

This is an AGP version '8.0' breaking change that changes the build
option default values, see build failure below:

PS: The alternative would be to disable the default, that is, having the
'android.nonFinalResIds' disabled, and not resolve all violating such
res references. However, because this didn't require a significant
amount extra work, nor much testing is needed, it is best not to ignore
this default so as to force this behavior going forward.

------------------------------------------------------------------------

> Task :WooCommerce:compileWasabiDebugKotlin FAILED
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/model/ProductCategory.kt:21:63
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/jetpack/JetpackInstallProgressDialog.kt:35:43
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/jetpack/JetpackInstallProgressDialog.kt:36:39
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt:138:49
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt:139:51
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt:140:48
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt:142:51
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt:143:53
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/main/kotlin/com/woocommerce/
android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt:144:50
Const 'val' initializer should be a constant value
✅ Build time report of 2m 57s has been received by Tracks.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':WooCommerce:compileWasabiDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner
 .GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

------------------------------------------------------------------------

Explanation: "Starting with AGP 8.0, the default values for these flags
have changed to improve build performance. AGP 8.0 generates R classes
with non-final fields by default."

For more info see: https://developer.android.com/build/releases/
gradle-plugin#default-changes

ParaskP7 committed:
Build: Upgrade gradle to 8.1.1 (2/2)

Release Notes: https://docs.gradle.org/8.1.1/release-notes.html
Command: ./gradlew wrapper --gradle-version=8.1.1
--distribution-type=all

------------------------------------------------------------------------

As per the previous 87b30ea64066817ffb76fc3378f7fb8d12c81581 commit and
its description, this change is a follow-up upgrade, which fully
completes this Gradle upgrade.

ParaskP7 committed:
Analysis: Suppress new api lint warning for context ext

Warning Message: "Call requires API level 30 (current min is 24):
android.content.Context#getDisplay"

Explanation: "This check scans through all the Android API calls in the
application and warns about any calls that are not available on all
versions targeted by this application (according to its minimum SDK
attribute in the manifest).

If you really want to use this API and don't need to support older
devices just set the minSdkVersion in your build.gradle or
AndroidManifest.xml files.

If your code is deliberately accessing newer APIs, and you have ensured
(e.g. with conditional execution) that this code will only ever be
called on a supported platform, then you can annotate your class or
method with the @TargetApi annotation specifying the local minimum SDK
to apply, such as @TargetApi(11), such that this check considers 11
rather than your manifest file's minimum SDK as the required API level.

If you are deliberately setting android: attributes in style
definitions, make sure you place this in a values-vNN folder in order to
avoid running into runtime conflicts on certain devices where
manufacturers have added custom attributes whose ids conflict with the
new ones on later platforms.

Similarly, you can use tools:targetApi="11" in an XML file to indicate
that the element will only be inflated in an adequate context."

------------------------------------------------------------------------

This Lint warning is suppressed because a check to
'SystemVersionUtils.isAtLeastR()' is already present, which guards from
accidentally using this API on devices lower are not on Android R
(API 30). It is just that Lint cannot recognize this custom
'SystemVersionUtils.isAtLeastR()' utility function to then stop
complaining about incorrect usage of this API on devices lower than
Android R (API 30).

ParaskP7 committed:
Build: Resolve all violating non final res references for android test

This is an AGP version '8.0' breaking change that changes the build
option default values, see build failure below:

PS: The alternative would be to disable the default, that is, having the
'android.nonFinalResIds' disabled, and not resolve all violating such
res references.

------------------------------------------------------------------------

> Task :WooCommerce:compileWasabiDebugAndroidTestKotlin FAILED
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/TabNavComponent.kt:12:37
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/TabNavComponent.kt:13:35
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/TabNavComponent.kt:14:37
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/TabNavComponent.kt:15:38
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/EmailAddressScreen.kt:8:41
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/EmailAddressScreen.kt:9:33
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/MagicLinkScreen.kt:8:41
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/PasswordScreen.kt:9:36
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/PasswordScreen.kt:10:33
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/SiteAddressScreen.kt:8:40
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/SiteAddressScreen.kt:9:33
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/WelcomeScreen.kt:10:34
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/login/WelcomeScreen.kt:11:40
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/moremenu/MoreMenuScreen.kt:16:36
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/mystore/MyStoreScreen.kt:14:30
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/mystore/StatsComponent.kt:8:37
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/mystore/settings/BetaFeaturesScreen.kt:8:43
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/mystore/settings/SettingsScreen.kt:14:33
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/mystore/settings/SettingsScreen.kt:15:42
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/mystore/settings/SettingsScreen.kt:16:36
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/CardReaderPaymentScreen.kt:8:36
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/CustomerDetailsScreen.kt:18:36
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/CustomerDetailsScreen.kt:19:38
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/CustomerDetailsScreen.kt:20:29
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/CustomerDetailsScreen.kt:21:34
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/OrderListScreen.kt:18:31
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/OrderListScreen.kt:19:31
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/OrderListScreen.kt:20:35
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/OrderListScreen.kt:21:41
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/PaymentSelectionScreen.kt:8:33
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:19:43
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:20:32
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:21:37
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:22:34
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:23:34
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:24:44
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:25:35
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:26:43
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:27:38
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/SingleOrderScreen.kt:28:29
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:23:35
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:24:42
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:25:38
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:26:33
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:27:40
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:28:32
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:29:35
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:30:36
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:31:37
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:32:38
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:33:37
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:34:43
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:35:29
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/orders/UnifiedOrderScreen.kt:36:45
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/products/ProductListScreen.kt:18:31
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/products/SingleProductScreen.kt:15:46
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/reviews/ReviewsListScreen.kt:15:31
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/reviews/SingleReviewScreen.kt:14:40
Const 'val' initializer should be a constant value
e: file:///Users/.../WooCommerce/src/androidTest/kotlin/com/woocommerce/
android/e2e/screens/shared/FilterScreen.kt:15:29
Const 'val' initializer should be a constant value
✅ Build time report of 0m 46s has been received by Tracks.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task
 ':WooCommerce:compileWasabiDebugAndroidTestKotlin'.
> A failure occurred while executing
 org.jetbrains.kotlin.compilerRunner
 .GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

------------------------------------------------------------------------

Explanation: "Starting with AGP 8.0, the default values for these flags
have changed to improve build performance. AGP 8.0 generates R classes
with non-final fields by default."

For more info see: https://developer.android.com/build/releases/
gradle-plugin#default-changes

ParaskP7 committed:
Build: Upgrade gradle to 8.2.1

Release Notes: https://docs.gradle.org/8.2.1/release-notes.html
Command: ./gradlew wrapper --gradle-version=8.2.1
--distribution-type=all

ParaskP7 committed:
Test: Fix gson related test failures on site picker view model test

Failure Message: "com.google.gson.JsonIOException: Failed making field
'java.lang.Throwable#detailMessage' accessible; either increase its
visibility or write a custom TypeAdapter for its declaring type.

...

Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make
field private java.lang.String java.lang.Throwable.detailMessage
accessible: module java.base does not "opens java.lang" to unnamed
module @7b36aa0c"

Explanation: "This is most probably due to the fact that support for
throwable/exception no longer works since Java 17."

For more info see: https://github.com/google/gson/issues/2352

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into build/upgrade-gradle-to-8.2.1-and-agp-to-8.0.2

oguzkocer committed:
Add .java-version file, setting the project JDK version to 17.0

oguzkocer reviewed (approved):
Looks good to me. Approving the PR pending testing.

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into build/upgrade-gradle-to-8.2.1-and-agp-to-8.0.2

 Conflicts:
	WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/screens/orders/UnifiedOrderScreen.kt

ParaskP7 committed:
Deps: Update media picker version to pr hash

This 'Media Picker' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.0.2'
respectively.

Media Picker PR: https://github.com/wordpress-mobile/
WordPress-MediaPicker-Android/pull/69

This step is required in order to check that these 'Media Picker'
related changes work as expected for WCAndroid.

ParaskP7 committed:
Deps: Update login version to pr hash

This 'Login' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.0.2'
respectively.

Login PR: https://github.com/wordpress-mobile/
WordPress-Login-Flow-Android/pull/119

This step is required in order to check that these 'Login'
related changes work as expected for WCAndroid.

ParaskP7 committed:
Deps: Update fluxc version to pr hash

This 'FluxC' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.0.2'
respectively.

FluxC PR: https://github.com/wordpress-mobile/
WordPress-FluxC-Android/pull/2772

This step is required in order to check that these 'FluxC'
related changes work as expected for WCAndroid.

ParaskP7 committed:
Build: Replace dependency substitution substitute with method with using

This is done because otherwise local builds can't be configured due a
Gradle API update that is causing a failures such as the below for the
'FluxC' composite build:

------------------------------------------------------------------------

FAILURE: Build failed with an exception.

* Where:
Script '/Users/.../config/gradle/include_builds.gradle' line: 19

* What went wrong:
No signature of method: org.gradle.api.internal.artifacts.ivyservice
 .dependencysubstitution.DefaultDependencySubstitutions$1.with()
 is applicable for argument types: (org.gradle.internal.component.local
 .model.DefaultProjectComponentSelector)
 values: [project :WordPress-FluxC-Android:fluxc]
Possible solutions: with(groovy.lang.Closure), with(boolean,
 groovy.lang.Closure), wait(), wait(long), find(), wait(long, int)
> No signature of method: org.gradle.api.internal.artifacts.ivyservice
 .dependencysubstitution.DefaultDependencySubstitutions$1.with()
 is applicable for argument types: (org.gradle.internal.component.local
 .model.DefaultProjectComponentSelector)
 values: [project :WordPress-FluxC-Android:fluxc]
  Possible solutions: with(groovy.lang.Closure), with(boolean,
   groovy.lang.Closure), wait(), wait(long), find(), wait(long, int)

------------------------------------------------------------------------

ParaskP7 commented:
Awesome @oguzkocer , thanks for adding JDK17 support to this repo, then reviewing and tentatively approving the changes, pending some extra testing! 🙇 ❤️ 🚀 

FYI: I went ahead and updated all 3 library references to PR hash (for testing purposes) and marked this PR as `Ready for review`. However, I am not removing the `status: do not merge` label just yet, as I want to wait for us to first merge all 3 library PRs into their `trunk` and then update all 3 library references to `trunk` hash before we go and do that. Wdyt? 🤔

oguzkocer commented:
> FYI: I went ahead and updated all 3 library references to PR hash (for testing purposes) and marked this PR as `Ready for review`. However, I am not removing the `status: do not merge` label just yet, as I want to wait for us to first merge all 3 library PRs into their `trunk` and then update all 3 library references to `trunk` hash before we go and do that. Wdyt? 🤔 

Sounds good to me 👍

oguzkocer commented on build.gradle:
```diff
@@ -112,8 +112,8 @@ ext {
-    wordPressLoginVersion = '1.1.0'
+    mediapickerVersion = '69-c5ef966709a7808f8780ddfa28ed19d0b0a2b607'
+    wordPressLoginVersion = '119-864c02e4f7b59d7d83cd5bc7a6dc92fb41595c00'
     aboutAutomatticVersion = '0.0.6'
```
Should we update this to the AGP update PR as well? There is also `aztecVersion` & `wordPressUtilsVersion`, but these are quite a bit behind the current `trunk`, so we probably don't want to update them. However, if the `aboutAutomatticVersion` update works without code changes, I think it'd be good to include that one before that also gets too outdated.

ParaskP7 replied:
Not sure about this @oguzkocer , I only dealt with any [composite related](https://github.com/woocommerce/woocommerce-android/blob/trunk/local-builds.gradle-example#L17-L19) builds, ignoring any other library, just to avoid any additional surprises, as doing this extra work isn't required to unblock upgrading Gradle & AGP for this client. 🤔

PS: Same applies to `aztecVersion` & `wordPressUtilsVersion`.

-----

Suggestion (💡): If you like this idea, after we merge this work, on multiple separate PRs, per such library update, we can do that, and also, take this opportunity to enabled composite builds on them. Then, we could have someone from the product team review that change, test it thoroughly, and finally, approve and merge. Wdyt?

oguzkocer replied:
I don't think we want to add more composite builds unless the team is going to use it. The composite builds add extra complication to our process and that's worth it when it's actively being used, but I'd refrain from adding new libraries until there is a request for it.

Otherwise, I am on board with opening follow up PRs to update the library versions. 👍

ParaskP7 replied:
Great, I actually agree on this comment of yours! 🎯 

Thus, for that matter, IMHO, we shouldn't deal with this `aboutAutomatticVersion` update, or any such library update for that matter. I just suggested that because I thought this is what was the main purpose of your comment, since the Gradle & AGP upgrade on any of those libraries are irrelevant to a client, that is, unless they are to be used as composite builds, is it not? 😊

oguzkocer replied:
We actually had some issues when the library's AGP version was ahead of the client, such as between `WooCommerce-Shared` & `WCAndroid`. Although this wouldn't be the case here, since we are already updating the client, I think the specifically AGP `8.0` update deserves extra precautions from us. That's why I suggested the update, so we can eliminate one more possibility of a build breaking when a developer tries to update the library.

Having said that, I don't see any reason for such an issue to occur in this particular case. Also, if we are going to follow it up with a PR ourselves, we'd still be the ones dealing with the build issue, so it doesn't matter much.

So, I wouldn't say the library AGP updates are irrelevant to the client, even if there is no composite build. However, I think we are safe in this case. Hope that clarifies my thinking a bit.

ParaskP7 replied:
> We actually had some issues when the library's AGP version was ahead of the client, such as between WooCommerce-Shared & WCAndroid. Although this wouldn't be the case here, since we are already updating the client, I think the specifically AGP 8.0 update deserves extra precautions from us. That's why I suggested the update, so we can eliminate one more possibility of a build breaking when a developer tries to update the library.

Thanks for the extra clarification on that @oguzkocer , much appreciated and TIL about `WooCommerce-Shared` & `WCAndroid`.

> Having said that, I don't see any reason for such an issue to occur in this particular case. Also, if we are going to follow it up with a PR ourselves, we'd still be the ones dealing with the build issue, so it doesn't matter much.

👍 

> So, I wouldn't say the library AGP updates are irrelevant to the client, even if there is no composite build. However, I think we are safe in this case. Hope that clarifies my thinking a bit.

That clarifies, thanks! 👍 

For what is worth, I still feel that we are safer to keep `aboutAutomatticVersion` at [0.0.6](https://github.com/Automattic/about-automattic-android/releases/tag/0.0.6) (`Jul 7, 2022`) and not update it to the most recent (to be created) `1.2.0` update, with [1.1.0](https://github.com/Automattic/about-automattic-android/releases/tag/1.1.0) (`Mar 24, 2023`) being the current most recent release. My fear is that this library being outdated for already 1 year+ it might break the current build, or even worse, have some kind of a runtime crash that we would need to deal with afterwards.

oguzkocer replied:
That's totally fine. We can do it as a follow up PR as you suggested.

ParaskP7 committed:
Deps: Update fluxc version to pr hash (v2)

This 'FluxC' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.0.2'
respectively.

FluxC PR: https://github.com/wordpress-mobile/
WordPress-FluxC-Android/pull/2772

This step is required in order to check that these 'FluxC'
related changes work as expected for WCAndroid.

------------------------------------------------------------------------

This is done to make sure that the new fluxc pr hash is pointing to the
most recent 'trunk' version of 'FluxC'.

oguzkocer commented:
@wzieba Once this PR is merged, it'll likely going to break the composite build for `WooCommerce-Shared`. I looked into updating Gradle & AGP for `WooCommerce-Shared` to avoid that, but it's causing Kotlin versioning related issues in `react-native-gradle-plugin`. There might be ways around this, but it looks like AGP version for `react-native` just got updated in https://github.com/facebook/react-native/commit/6cd1da94b83eaac959624aa1e688bac6b9ea036f and set to be released in the next version.

We are probably at least a couple days away from merging these PRs, so I think it's best for us to wait before we update AGP for `WooCommerce-Shared` until a new version of `react-native` is released. You should be able to temporarily keep using composite builds if you don't merge `trunk` to your branch where you're integrating `WooCommerce-Shared`, so you might not even get disrupted if they release the new version quickly.

I'll keep an eye on `react-native` releases to take care of this, but please feel free to reach out if I miss the release, or if this becomes a blocker for you.

wzieba commented:
Thank you for the ping and update @oguzkocer 👍 . Sure, I understand current circumstances. 

Just want add that the lack of included builds for some time won't be a blocker for `WooCommerce-Shared` as we now mostly work on the library itself. It's possible to do this by passing auth credentials via environmental variables added in https://github.com/woocommerce/WooCommerce-Shared/pull/66

oguzkocer commented:
Thank you for understanding @wzieba and it's great that composite builds are not a necessity at this point! Please let us know if the situation changes and we'll be sure to prioritize it.

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into build/upgrade-gradle-to-8.2.1-and-agp-to-8.0.2

 Conflicts:
	build.gradle

ParaskP7 committed:
Build: Upgrade agp to 8.1.0

Release Notes: https://developer.android.com/build/releases/
gradle-plugin#8-1-0

ParaskP7 committed:
Deps: Update media picker version to pr hash (v2)

This 'Media Picker' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.1.0'
respectively.

Media Picker PR: https://github.com/wordpress-mobile/
WordPress-MediaPicker-Android/pull/69

This step is required in order to check that these 'Media Picker'
related changes work as expected for WCAndroid.

ParaskP7 committed:
Deps: Update login version to pr hash (v2)

This 'Login' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.1.0'
respectively.

Login PR: https://github.com/wordpress-mobile/
WordPress-Login-Flow-Android/pull/119

This step is required in order to check that these 'Login'
related changes work as expected for WCAndroid.

ParaskP7 committed:
Deps: Update fluxc version to pr hash (v3)

This 'FluxC' PR hash updates the library to that branch version
where 'Gradle & AGP' is also upgraded to version '8.2.1 & 8.1.0'
respectively.

FluxC PR: https://github.com/wordpress-mobile/
WordPress-FluxC-Android/pull/2772

This step is required in order to check that these 'FluxC'
related changes work as expected for WCAndroid.

oguzkocer committed:
Update wordPressLoginVersion to 1.4.0

oguzkocer committed:
Update mediapickerVersion to 0.2.0

oguzkocer committed:
Update fluxCVersion to 2.39.0

oguzkocer committed:
Update automatticTracksVersion to 3.2.0
